kurl_proxy/cmd/main.go

```diff
@@ -249,6 +249,14 @@ func getHttpsServer(upstream *url.URL, tlsSecretName string, secrets corev1.Secr
 			return
 		}
 
+		hostString, success := c.GetPostForm("hostname")
+		if success != true {
+			log.Println("Invalid hostname")
+			c.AbortWithStatus(http.StatusInternalServerError)
```

Comment:
Why not 400?

Reply:
could this default to whatever hostname is on the request if not set? I know for dev most people leave the hostname blank.

Reply:
So I'm not totally certain when this would fail.  Its definitely not when the user leaves the hostname field blank (I did test that scenario).  So for now, internal error seemed to make sense to me.